### ISSUE TITLE
Use NFS mounts instead of osxfs

### ISSUE THREAD

**Issue by fuerst** [drud/ddev#976]
As you may know or experience by yourself volumes mounted by Docker for Mac's _osxfs_ provide slow I/O performance. There is a lengthy topic about this at the [Docker Forum](https://forums.docker.com/t/file-access-in-mounted-volumes-extremely-slow-cpu-bound/8076). 

Even if osxfs seems to do well reading files using the _cached_ configuration it still is slow at writes compared to NFS. Also - as I understand it - _cached_ may have it's drawbacks like delayed synchronisation of host changes to containers.

Are there any plans to use NFS mounts instead of osxfs which seems to be possible since Docker for Mac (around) 18.04? Or is it just not worth it because _cached_ already works well enough?

**Comment by cweagans** [comment]
Oh, I didn't know Docker for Mac shipped NFS support! https://medium.com/@sean.handley/how-to-set-up-docker-for-mac-with-native-nfs-145151458adc has some interesting setup stuff to look into. In any case, it might be worth playing with. Thanks for the heads up!

**Comment by tmotyl** [comment]
I'm also hit by poor osxfs performance on maxos. Any alternative like nfs or rsync base synch would be very appreciated.

**Comment by tmotyl** [comment]
related issue: https://github.com/drud/ddev/issues/835

**Comment by wizonesolutions** [comment]
The solution in #835 (webcache feature, `webcache_enabled: true` in config seems to be solving this for a lot of use cases?

**Comment by fuerst** [comment]
Recently I used docker-machine + VirtualBox + NFS and was amazed by the performance of file system operations. A big Drupal 7 site runs faster than using Docker Desktop. Both on macOS 10.14

Currently I'm running the same site at Docker Desktop + webcache and it is good. Startup needs more time though: 14 sec for 0.5 GB data that needs to be copied to the cache. What if you got many GB or i.e. just forget to delete a SQL dump file before starting DDEV?

I would prefer NFS if I had a choice because I think it has lesser overhead and is still faster compared to webcache (Unisono).

**Comment by rfay** [comment]
NFS is nearly ready for community testing. And it works on Windows with Docker for Windows, Windows with Docker Toolbox, macOS, and Linux.

**Comment by wizonesolutions** [comment]
It will be interesting to compare the performance. NFS is still definitely not native-speed, whereas webcache basically is. There is definitely the overhead of initial copying, so I guess it depends on how often one is restarting the containers/switching between projects.

**Comment by cweagans** [comment]
I did this comparison a year or two ago, and webcache was still ~3-4x faster. You also lose out on inotify support with NFS (fs events are not propagated through the NFS mount).

**Comment by wizonesolutions** [comment]
Yes, fsevents is another thing I was thinking about. I wish we could have
NFS while webcache was warming up or something, and then switch :D
progressive volume enhancement.
